backend/src/main/java/fk/prof/backend/Configuration.java

```diff
@@ -0,0 +1,401 @@
+                                         @JsonProperty("parser") ParserConfig parserConfig) {
+            this.verticleCount = verticleCount;
+            this.reportLoad = getOrDefault(reportLoad, true);
+            this.parserConfig = parserConfig;
+        }
+
+        @NotNull
+        public final Integer verticleCount;
+        @NotNull
+        public final Boolean reportLoad;
+        @NotNull
+        @Valid
+        public final ParserConfig parserConfig;
+
+        public static class ParserConfig {
+            @JsonCreator
+            public ParserConfig(@JsonProperty("recordingheader.max.bytes") Integer recordingHeaderMaxSizeBytes,
+                                @JsonProperty("wse.max.bytes") Integer wseMaxSizeBytes) {
+                this.recordingHeaderMaxSizeBytes = recordingHeaderMaxSizeBytes;
+                this.wseMaxSizeBytes = wseMaxSizeBytes;
+            }
+
+            @NotNull
+            public final Integer recordingHeaderMaxSizeBytes;
+            @NotNull
+            public final Integer wseMaxSizeBytes;
+        }
+    }
+
+    public static class LeaderElectionVerticleConfig {
+        @JsonCreator
+        public LeaderElectionVerticleConfig(@JsonProperty("aggregation.enabled") Boolean aggregationEnabled,
+                                            @JsonProperty("leader.watching.path") String leaderWatchPath,
+                                            @JsonProperty("leader.mutex.path") String leaderMutexPath,
+                                            @JsonProperty("kill.behavior") KillBehavior killBehaviour) {
+            this.aggregationEnabled = getOrDefault(aggregationEnabled, false);
+            this.leaderWatchPath = leaderWatchPath;
+            this.leaderMutexPath = leaderMutexPath;
+            this.killBehaviour = getOrDefault(killBehaviour, KillBehavior.DO_NOTHING);
+        }
+
+        @NotNull
+        public final Boolean aggregationEnabled;
+        @NotNull
+        public final String leaderWatchPath;
+        @NotNull
+        public final String leaderMutexPath;
+        @NotNull
+        public final KillBehavior killBehaviour;
+    }
+
+    public static class CuratorConfig {
+        @JsonCreator
+        public CuratorConfig(@JsonProperty("connection.url") String connectionUrl,
+                             @JsonProperty("namespace") String namespace,
+                             @JsonProperty("connection.timeout.ms") Integer connectionTimeoutMs,
+                             @JsonProperty("session.timeout.ms") Integer sessionTineoutMs,
+                             @JsonProperty("max.retries") Integer maxRetries) {
+            this.connectionUrl = connectionUrl;
+            this.namespace = namespace;
+            this.connectionTimeoutMs = connectionTimeoutMs;
+            this.sessionTineoutMs = sessionTineoutMs;
+            this.maxRetries = maxRetries;
+        }
+
+        @NotNull
+        public final String connectionUrl;
```

Comment:
As we discussed in userapi PR, lets keep configuration immutable. All public variables should be replaced with private setters and public getters.
I understand it is a bit more verbosity but protects us from mistaken config assignments (which are by default global in nature) and maintains the contract that what I configure in a file is "always" what I get in code